R/load_package.r

```diff
@@ -1,15 +1,42 @@
 #' Load a package and install it if necessary
 #' 
 #' @name load_package
-#' @param name Name of package
+#' @param name character. Name of package.
+#' @param verbose logical. Whether or not to announce each installation.
 #' @examples
 #' \dontrun{
-#' load_package('glmnet')
+#" load_package("glmnet")
 #' }
-load_package <- function(name) {
-  if (!require(name, character.only = TRUE)) {
-    install.packages(name, dependencies = TRUE)
-    if (!require(name, character.only = TRUE))
-      stop(paste('Package', name, 'not found'))
+load_package <- function(name, verbose = FALSE) {
+  if (package_is_installed(name)) {
+    if (isTRUE(verbose)) { message(name, " already installed.") }
+    return(TRUE)
   }
+  if (is_github_package(name)) {
+    remote <- "github"
+    if (isTRUE(verbose)) { announce(name, remote) }
```

Comment:
Can you add a function `ensure_devtools_installed` in utils.R which errors if devtools isn't installed? Also, if you are going to be using devtools here you should add it to the the Suggest section in the DESCRIPTION file.